packages/app-lib/src/api/profile/mod.rs

```diff
@@ -775,13 +775,30 @@ pub async fn run_credentials(
         ))
     })?;
 
+    let java_args = profile
+    .java
+    .as_ref()
+    .and_then(|it| it.extra_arguments.as_ref())
+    .unwrap_or(&settings.custom_java_args);
+
     let pre_launch_hooks =
         &profile.hooks.as_ref().unwrap_or(&settings.hooks).pre_launch;
     if let Some(hook) = pre_launch_hooks {
-        // TODO: hook parameters
-        let mut cmd = hook.split(' ');
+        // TODO: Improve Hook parameters
+        
+        let full_path = path.get_full_path().await?;
+        let jre_key = get_optimal_jre_key(&path).await.unwrap();
```

Comment:
```suggestion
        let jre_key = get_optimal_jre_key(&path).await?;
```
Here and in other places use [`?`](https://doc.rust-lang.org/rust-by-example/std/result/question_mark.html) instead of `.unwrap()`